python/lsst/ts/atdometrajectory/dome_trajectory.py

```diff
@@ -217,6 +341,32 @@ async def handle_summary_state(self):
         if not self.summary_state == salobj.State.ENABLED:
             self.move_dome_azimuth_task.cancel()
             await self.evt_followingMode.set_write(enabled=False)
+        if self.disabled_or_enabled:
+            if self.report_vignetted_task.done():
+                self.report_vignetted_task = asyncio.create_task(
+                    self.report_vignetted_loop()
+                )
+        else:
+            self.report_vignetted_task.cancel()
+
+    async def report_vignetted_loop(self):
+        """Poll ATMCS and ATDome topics to report telescopeVignetted event."""
+        self.log.info("report_vignetted_loop begins")
+        try:
+            while True:
+                azimuth = self.compute_vignetted_by_azimuth()
```

Comment:
Both calls require (sooner or later) telescope elevation. Without telescope elevation, you are lost anyway (it really doesn't depend on shutter open/close, as you cannot tell and it's for me irrelevant if you report VIGNETTED or UNKNOWN). It will me a bit better to move elevation encoders retrival to there and pass elevation as parameter to compute_..function - that might faciliate possible testing among other things. Also if you later decide you would like to check elevation is recent (with get you are getting history without any confirmation that this is current encoder readout), you will have only one place to change the code.

Reply:
Yes, I thought about this. The issue is that there might not be any data at all available. But I suppose I know the answer is UNKNOWN if there is not. I'll think about it.